source/updateCheck.py

```diff
@@ -242,17 +283,78 @@ def onDownloadButton(self, evt):
 		DonateRequestDialog(gui.mainFrame, self._download)
 
 	def _download(self):
-		if self.isInstalled:
-			UpdateDownloader(self.urls, fileHash=self.fileHash).start()
-		else:
-			os.startfile(self.urls[0])
+		UpdateDownloader(self.urls, fileHash=self.fileHash).start()
 		self.Destroy()
 
 	def onLaterButton(self, evt):
 		state["dontRemindVersion"] = None
 		saveState()
 		self.Close()
 
+class UpdateAskInstallDialog(wx.Dialog):
+
+	def __init__(self, parent, destPath, version):
+		self.destPath=destPath
+		self.version = version
+		storeUpdatesDirWritable=os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir)
+		# Translators: The title of the dialog asking the user to Install an NVDA update.
+		super(UpdateAskInstallDialog, self).__init__(parent, title=_("NVDA Update"))
+		mainSizer = wx.BoxSizer(wx.VERTICAL)
+		if not storeUpdatesDirWritable:
+			# Translators: A message indicating that an updated version of NVDA can either be installed or discarded.
+			message=_("You can either install it now, or discard the update and download it again at a later time.")
+		else:
+			# Translators: A message indicating that an updated version of NVDA can be installed, postponed or discarded.
+			message=_("You can install it now, or install it later from the NVDA menu or the exit dialog, if enabled.\n")
+		mainSizer.Add(wx.StaticText(self, label=
+			# Translators: A message indicating that an updated version of NVDA is ready to be installed.
+			_("Update is ready to install.\n")+message+

```

Comment:
This building up of messages has a risk that it won't make sense once translated. I also think these labels could be simplified.  I think it is adequate to simply set the label as "Update is ready to install" and rely on the presence of the buttons to indicate what options are available.